yt_dlp/extractor/weverse.py

```diff
@@ -46,8 +46,8 @@ def _perform_login(self, username, password):
             'x-clog-user-device-id': str(uuid.uuid4()),
         }
         check_username = self._download_json(
-            f'{self._ACCOUNT_API_BASE}/signup/email/status', None,
-            note='Checking username', query={'email': username}, headers=headers)
+            f'{self._ACCOUNT_API_BASE}/signup/email/status', None, note='Checking username',
+            query={'email': username}, headers=headers, expected_status=(200, 400, 404))
         if not check_username.get('hasPassword'):
```

Comment:
It's not necessary to add `200` to `expected_status`

I think we should also use traversal here, and make the var name make more sense
```py
        valid_username = traverse_obj(self._download_json(
            f'{self._ACCOUNT_API_BASE}/signup/email/status', None, note='Checking username',
            query={'email': username}, headers=headers, expected_status=(400, 404)), 'hasPassword')
        if not valid_username:
```